lightning/src/util/ser_macros.rs

```diff
@@ -186,12 +184,15 @@ macro_rules! decode_tlv {
 	($reader: expr, $field: ident, (option: $trait: ident $(, $read_arg: expr)?)) => {{
 		$field = Some($trait::read(&mut $reader $(, $read_arg)*)?);
 	}};
-	($reader: expr, $field: ident, (tlv_record, $fieldty:tt)) => {{
+	($reader: expr, $field: ident, (tlv_record, ($fieldty:ty, $wrapper:ident))) => {{
 		$field = {
-			let field: encoded_tlv_record_type!($fieldty) = ser::Readable::read(&mut $reader)?;
+			let field: $wrapper<$fieldty> = ser::Readable::read(&mut $reader)?;
```

Comment:
Note that I can change the type declaration to `$wrapper<_>`.

Reply:
The issue is there isn't a clean way of _not_ passing `$fieldty` through. So I figured we might as well use it instead of having the compiler infer it.